src/adapter/src/coord/appends.rs

```diff
@@ -240,18 +240,23 @@ impl<S: Append + 'static> Coordinator<S> {
                         },
                 } => {
                     for WriteOp { id, rows } in writes {
-                        // If the table that some write was targeting has been deleted while the write was
-                        // waiting, then the write will be ignored and we respond to the client that the
-                        // write was successful. This is only possible if the write and the delete were
-                        // concurrent. Therefore, we are free to order the write before the delete without
-                        // violating any consistency guarantees.
+                        // If the table that some write was targeting has been deleted while the
+                        // write was waiting, then the write will be ignored and we respond to the
+                        // client that the write was successful. This is only possible if the write
```

Comment:
Might this be weird for a person observing through a SINK or TAIL?

Reply:
yes, that's a good point. If someone is executing a tail, there's a possibility that we respond with an OK to a client who performs an insert and the person executing the tail won't see it.

This is actually existing behavior but probably not very good. Now that DDL is combined with writes, we should be able to remove this condition. Before actually deleting a table, we'd executing all pending writes which means that this scenario is impossible. I'd probably prefer to do that in a follow up though. I'll open an issue.